add test for velocity verlet

rlouf opened the pull request:
This PR add tests for the velocity verlet algorithm implementation

rlouf commented on tests/test_integrators.py:
```diff
@@ -0,0 +1,92 @@
+    step_size = aet.scalar("step_size")
+    energy = potential(q)
+    energy_grad = aesara.grad(energy, q)
+    trajectory, _ = aesara.scan(
+        fn=step_fn,
+        outputs_info=[q, p, energy, energy_grad],
+        non_sequences=step_size,
+        n_steps=n_steps,
+    )
```
This fails with the following error message:

```
ValueError: Scan cannot parse the return value of your lambda expression, which is: (Elemwise{add,no_inplace}.0, Elemwise{sub,no_inplace}.0, Sum{acc_dtype=float64}.0, [Elemwise{mul}.0])
```

and I cannot figure out why after reading the documentation and the code.

brandonwillard reviewed (changes requested):
This change breaks `mypy` checking.  Did you set up `pre-commit`, because it should've prevented this commit?

brandonwillard commented:
@rlouf, I've added a commit that adjust the use of `aesara.function` and `aesara.scan` so that the evaluations can be performed.

rlouf commented:
@brandonwillard I fixed the few bugs that were left, added tests to check that we integrate the correct trajectory, and a test on 2D state vectors. Good on my end.
